Add support for playerctl, allowing tmux-now-playing to support (e.g.) Spotify under Linux

renanbirck opened the pull request:
This pull request adds support for playerctl, allowing players that follow the MPRIS standard (a list is available at https://wiki.archlinux.org/title/MPRIS) to display now-playing in the tmux status bar using your plug-in.

Very hacky, and can be tested more. But seems to work with Spotify at least.

renanbirck committed:
feat(players, scripts): an ugly, hacky playerctl approach

	tested with Spotify only, for now, only under Linux.
	Probably breaks easily.

renanbirck committed:
fix(playerctl): add 1 second of sleep
	Hacky, but... it works, and updates are smoother.

renanbirck committed:
feat(playerctl): keybindings now work

spywhere commented on README.md:
```diff
@@ -13,6 +13,9 @@ Showing currently playing track in tmux status bar with music controls
   - iTunes
 - `mpd` ([Music Player Daemon](https://www.musicpd.org)) through `nc` (netcat)
 - [`nowplaying-cli`](https://github.com/kirtan-shah/nowplaying-cli)
+- (Linux) anything that supports MPRIS through `playerctl`
```
nitpick: can you update this line to align with `osascript` above? Would be best if we could link to the `playerctl` home page/repo as well.

spywhere commented on players/playerctl.sh:
```diff
@@ -0,0 +1,62 @@
+
+is_running() {
+
+  if (playerctl status | grep "No players found") then 
```
suggestion: put `-q` for grep to not produce any output.

spywhere commented on players/playerctl.sh:
```diff
@@ -0,0 +1,62 @@
+}
+
+is_playing() {
+  if ! is_running; then # Isn't running 
```
nitpick: no need for unnecessary comment

spywhere commented on players/playerctl.sh:
```diff
@@ -0,0 +1,62 @@
+}
+
+get_music_data() {
+
```
nitpick: trim empty line

spywhere commented on players/playerctl.sh:
```diff
@@ -0,0 +1,62 @@
+    status="paused"
+  fi
+
+  printf "%s\n%s\n%s\n%s\n%s\nSpotify" "$status" "$position" "$duration" "$artist" "$title"
```
suggestion: if we can't figure out the app from `playerctl`, then let's use `playerctl` in place of `Spotify`.

spywhere commented on players/playerctl.sh:
```diff
@@ -0,0 +1,62 @@
+  if test "$remote_command" = "pause"; then
+    playerctl play-pause
+  else
+    playerctl "$remote_command"
```
question: does `playerctl` support the rest of the remote commands?

For consistency, it should do nothing for any unsupported remote commands.

renanbirck replied:
Yes, playerctl supports the rest of the remote commands.

spywhere commented on README.md:
```diff
@@ -13,6 +13,9 @@ Showing currently playing track in tmux status bar with music controls
 - [`nowplaying-cli`](https://github.com/kirtan-shah/nowplaying-cli)
+- (Linux) anything that supports MPRIS through `playerctl`
+  - Spotify 
+  - probably can work with other players 
```
suggestion: if totally unsure, then let's keep it only for Spotify. If you think it should work, but cannot tested it, let's update this line to something like 'Any other players that `playerctl` supported'.

spywhere commented on players/playerctl.sh:
```diff
@@ -0,0 +1,62 @@
+source "$(dirname "$CURRENT_DIR")/scripts/helpers.sh"
+
+is_running() {
+
```
nitpick: trim empty line

spywhere reviewed (changes requested):
Thanks for the PR and sorry for the late review. Just small suggestions and comments, then it should be good to go.

renanbirck committed:
Merge branch 'spywhere:master' into master

renanbirck committed:
fix(playerctl, README): minor corrections
	As requested on the pull request review, made some minor corrections:

	- update README.md to make it more in-line with other players
	- use grep -q for grep to not product any output
	- remove unneeded comments and empty lines

renanbirck commented:
OK, I made the recommended corrections. I believe it should be good now.

spywhere commented:
Alright, thanks for the contribution. I'll revise a bit after then.
